src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +83,14 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList
+                .stream()
+                .filter(person -> person.getSex() == Person.Sex.WOMAN
+                        && person.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(List::stream)
```

Comment:
```suggestion
                .flatMap(person -> person.getCats().stream())
```
flatMap deletes people without cats